### ISSUE TITLE
Module version mismatch with distro version

### ISSUE THREAD

**Issue by bjakubski** [nrdvana/perl-Log-Any-Adapter-TAP#2]
According to Changes file:
> Version 0.003002
 >  - Fix distfile (no code changes, module version unchanged)

And indeed, while distro and META point to 0.003002 in TAP.pm there is still 0.003001.

This causes:
> $ cpanm Log::Any::Adapter::TAP@0.003002           
> ! Finding Log::Any::Adapter::TAP (== 0.003002) on cpanmetadb failed.
> Found Log::Any::Adapter::TAP 0.003001 which doesn't satisfy == 0.003002.

**Comment by nrdvana** [comment]
On 09/23/2015 09:28 AM, Bartosz Jakubski wrote:
>
> According to Changes file:
>
>     Version 0.003002
>
>       * Fix distfile (no code changes, module version unchanged)
>
> And indeed, while distro and META point to 0.003002 in TAP.pm there is 
> still 0.003001.
>

My intent was to prevent unnecessary upgrades and noise.  i.e. people 
re-installing the same code just because another package decided to 
depend on .003002


> This causes:
>
>     $ cpanm Log::Any::Adapter::TAP@0.003002
>
>     ! Finding Log::Any::Adapter::TAP (== 0.003002) on cpanmetadb failed.
>     Found Log::Any::Adapter::TAP 0.003001 which doesn't satisfy ==
>     0.003002.
>

But
   $ cpanm Log::Any::Adapter::TAP@0.003001
works as expected.  And if you use a tool to collect the dependencies of 
your code, it will detect 0.003001 rather than the dist version.  While 
uncommon, I think there are a lot of dists out there that contain 
versioned modules that differ from the dist version that contains them.  
Do you still feel that this needs fixed?

-Mike

**Comment by bjakubski** [comment]
I've seen a lot of new dist versions released containg only minor metadata changes. I don't think one has to worry about any noise that introduces, especially there is no risk of breaking any code.

I stumbled upon it when I installed latest version (it said 003002) and added that version to our metadata with installed perl module versions. Tool that installs it translated that to cpanm LAA::TAP@003002 and it failed
I think it is really nice to have consistent metadata :-)

BTW It does not matter much, but cpanm @003001 will fetch the tarball with the original 003001 (with the extra tar.gz files inside)
